commerce-order-web/src/main/java/com/liferay/commerce/order/web/internal/frontend/OrderItemFilterFactoryImpl.java

```diff
@@ -0,0 +1,58 @@
+/**
+ * Copyright (c) 2000-present Liferay, Inc. All rights reserved.
+ *
+ * This library is free software; you can redistribute it and/or modify it under
+ * the terms of the GNU Lesser General Public License as published by the Free
+ * Software Foundation; either version 2.1 of the License, or (at your option)
+ * any later version.
+ *
+ * This library is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+ * FOR A PARTICULAR PURPOSE. See the GNU Lesser General Public License for more
+ * details.
+ */
+
+package com.liferay.commerce.order.web.internal.frontend;
+
+import com.liferay.commerce.frontend.Filter;
+import com.liferay.commerce.frontend.FilterFactory;
+import com.liferay.portal.kernel.util.ParamUtil;
+
+import javax.servlet.http.HttpServletRequest;
+
+import org.osgi.service.component.annotations.Component;
+
+/**
+ * @author Alessio Antonio Rendina
+ */
+@Component(
+	immediate = true,
+	property = "commerce.data.provider.key=" + CommerceOrderItemClayTable.NAME,
+	service = FilterFactory.class
+)
+public class OrderItemFilterFactoryImpl implements FilterFactory {
+
+	@Override
+	public Filter create(HttpServletRequest httpServletRequest) {
+		OrderItemFilterImpl orderItemFilter = new OrderItemFilterImpl();
+
+		boolean advancedSearch = ParamUtil.getBoolean(
+			httpServletRequest, "advancedSearch");
+		boolean andOperator = ParamUtil.getBoolean(
+			httpServletRequest, "andOperator", true);
+		String name = ParamUtil.getString(httpServletRequest, "name");
+		String sku = ParamUtil.getString(httpServletRequest, "sku");
+
+		orderItemFilter.setAdvancedSearch(advancedSearch);
+		orderItemFilter.setAndOperator(andOperator);
+		orderItemFilter.setName(name);
+		orderItemFilter.setSku(sku);
+
+		String keywords = ParamUtil.getString(httpServletRequest, "keywords");
+
+		orderItemFilter.setKeywords(keywords);
```

Comment:
Is there a reason to separate out `keywords` like this? I noticed you did it once before, but it seems like the same type of thing as the other variables you're setting on the `orderItemFilter`, so I don't understand why it's separated out.

Reply:
Oh, I see.... `keywords` is an inherited field, huh? To me, it still makes sense to treat them all as the same type of thing. I think it'll be easier to maintain going forward if we do that, since we can use line breaks to indicate more important distinctions... but let me know what you think.